internals/overlord/overlord.go

```diff
@@ -49,6 +49,25 @@ var (
 	defaultCachedDownloads = 5
 )
 
+// Extension represents an extension of the Overlord.
+type Extension interface {
+	// ExtraManagers is called when building an Overlord.
+	ExtraManagers(o *Overlord) ([]StateManager, error)
+}
+
+// Options is the arguments passed to construct an Overlord.
+type Options struct {
+	// PebbleDir is the path to the pebble directory. It must be provided.
+	PebbleDir string
+	// RestartHandler is an optional structure to handle restart requests.
+	RestartHandler restart.Handler
+	// ServiceOutput is an optional output for the logging manager.
+	ServiceOutput io.Writer
+	// Extension is an optional struct defining additional components to add to
+	// the overlord.
```

Comment:
Yeah, the wording here is tricky, but I agree with @flotter it's weird/confusing to say "struct" when this is clearly an interface. (Which could be fulfilled by any type, even a `func` type, or another interface.)

I think the updated comment on `Extension.ExtraManagers` will help, and maybe after that we can change this to:

```suggestion
	// Extension allows extending the overlord with additional managers.
```